src/sinks/datadog/logs.rs

```diff
@@ -347,6 +383,14 @@ mod tests {
         crate::test_util::test_generate_config::<DatadogLogsConfig>();
     }
 
+    fn event_with_api_key(msg: &str, key: &str) -> Event {
+        let mut e = Event::from(msg);
+        e.as_mut_log()
+            .metadata_mut()
+            .merge(EventMetadata::new_with_datadog_api_key(key.to_string()));
+        e
```

Comment:
For new code, `Event::from` is now a pattern we'd like to avoid. Prefer this:
```suggestion
        let mut e = LogEvent::from(msg);
        e.metadata_mut()
            .merge(EventMetadata::new_with_datadog_api_key(key.to_string()));
        e.into()
```
(the last line can of course be `Event::from(e)` or `Event::Log(e)` if the code isn't coercing the type into an `Event`)

If this were anything but a test function, I would strongly suggest adding a `fn EventMetadata::set_datadog_api_key`, possibly automatically derived with `getset`.